hazelcast/src/main/java/com/hazelcast/internal/networking/AbstractChannel.java

```diff
@@ -14,26 +14,49 @@
  * limitations under the License.
  */
 
-package com.hazelcast.nio.tcp;
+package com.hazelcast.internal.networking;
 
-import com.hazelcast.internal.networking.Channel;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
 
 import java.io.IOException;
 import java.net.Socket;
 import java.net.SocketAddress;
 import java.nio.ByteBuffer;
-import java.nio.channels.SelectableChannel;
 import java.nio.channels.SocketChannel;
+import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
 
-public class PlainChannel implements Channel {
+import static com.hazelcast.util.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static java.util.Collections.newSetFromMap;
+import static java.util.concurrent.atomic.AtomicIntegerFieldUpdater.newUpdater;
+
+/**
+ * An abstract {@link Channel} implementation. This class is a pure implementation detail, the fact that it exposes some
+ * functionality like access to the socket channel is because the current Channel implementations need the SocketChannel.
+ */
+public abstract class AbstractChannel implements Channel {
+
+    private static final AtomicIntegerFieldUpdater<AbstractChannel> CLOSED = newUpdater(AbstractChannel.class, "closed");
 
     protected final SocketChannel socketChannel;
     private final ConcurrentMap<?, ?> attributeMap = new ConcurrentHashMap<Object, Object>();
+    private final Set<ChannelCloseListener> closeListeners
+            = newSetFromMap(new ConcurrentHashMap<ChannelCloseListener, Boolean>());
+    private final boolean client;
+    private volatile int closed;
```

Comment:
why not just a bool instead of int? is there any other value other than 1 and 0?